backends/dpdk/dpdkArch.cpp

```diff
@@ -474,6 +474,29 @@ bool CollectMetadataHeaderInfo::preorder(const IR::Type_Struct *s) {
        }
 */
 const IR::Node *AlignHdrMetaField::preorder(IR::Type_StructLike *st) {
+    if (st->is<IR::Type_Header>()) {
+        unsigned size_sum_so_far = 0;
+        bool all_hdr_field_aligned = true;
+        for (auto field : st->fields) {
+            if ((*field).type->is<IR::Type_Bits>()) {
+                auto t = (*field).type->to<IR::Type_Bits>();
+                auto width = t->width_bits();
+                size_sum_so_far += width;
+                if ((width & 0x7) != 0) {
+                    all_hdr_field_aligned = false;
+                }
+            }
```

Comment:
and if it's not bit<> you do nothing?

Reply:
it's always bit<> at this point.

Reply:
all fields will be of bit<> type at this point.

Reply:
that's what BUG is for.

Reply:
yes done